src/patterns/gen-ai/aws-qa-appsync-opensearch/index.ts

```diff
@@ -449,6 +466,13 @@ export class QaAppsyncOpensearch extends Construct {
         resources: ['*'],
       }),
     );
+    question_answering_function_role.addToPolicy(
```

Comment:
I think this should be done by the user if he deploys a SG model, not by this construct
You can add a code snippet to the documentation where we demonstrate how to deploy the idefics model using the SG construct, then the QA construct, and use the ```grantInvoke``` SG model method to give the correct permission to the QA construct lambda function

Reply:
done. added a prop for user to set sagemakerEnpointName in QA construct , if the Name is set then grant invoke to QA construct.